src/helpers/dialog.ts

```diff
@@ -143,6 +146,19 @@ export class Dialog<T> {
     });
   }
 
+  private _edgeDialog(): Promise<T> {
+    return new Promise((resolve, reject) => {
+      Office.context.ui.displayDialogAsync(this.url, { width: this.size.width$, height: this.size.height$ }, (result: Office.AsyncResult) => {
+        if (result.status === Office.AsyncResultStatus.Failed) {
+          reject(new DialogError(result.error.message, result.error));
+        }
+        else {
+          this._pollLocalStorageForToken(resolve, reject);
+        }
+      });
+    });
+  }
```

Comment:
Looks like there's a bug with this dialog on the Outlook desktop client (tested build 1908 & 1909). I use the fork this PR is based on, and my changes in my fork of the fork(🙄) seems to have fixed it.

The problem seem to be that the dialog isn't properly closed as far as `office.js` is concerned. It works as expected when using OWA in Edge, but not on the desktop client.

The scenario is the user is authenticated through the popup dialog right. And at some point the token is invalid and `offce-helpers` tries to open a new popup to authenticate the user again. But it receives an error from office.js with code 12007 "A dialog box is already opened from this host window."

These changes seems to have fixed the problem:
```diff
   private _edgeDialog(): Promise<T> {
     return new Promise((resolve, reject) => {
       Office.context.ui.displayDialogAsync(this.url, { width: this.size.width$, height: this.size.height$ }, (result: Office.AsyncResult) => {
+        let dialog = result.value as Office.DialogHandler;
+
         if (result.status === Office.AsyncResultStatus.Failed) {
           reject(new DialogError(result.error.message, result.error));
         }
         else {
-          this._pollLocalStorageForToken(resolve, reject);
+          const onToken = token => {
+            if (dialog) {
+              dialog.close();
+            }
+
+            return resolve(token);
+          };
+
+          this._pollLocalStorageForToken(onToken, reject);
         }
       });
     });
```